src/lavinmq/amqp/queue/message_store.cr

```diff
@@ -399,33 +413,85 @@ module LavinMQ
           @log.debug { "Loading #{@segments.size} segments" }
         end
         @segments.each do |seg, mfile|
-          count = 0u32
-          loop do
-            pos = mfile.pos
-            ts = IO::ByteFormat::SystemEndian.decode(Int64, mfile.to_slice(pos, 8))
-            break mfile.resize(pos) if ts.zero? # This means that the rest of the file is zero, so resize it
-            bytesize = BytesMessage.skip(mfile)
-            count += 1
-            next if deleted?(seg, pos)
+          next if mfile.size <= 4 # only got the schema version
+          # check if message count in segment is appended to the end
+          begin
+            count = read_msg_count(mfile)
+            @segment_msg_count[seg] = count
+            bytesize = mfile.size - 4 # minus schema version
```

Comment:
Subtract the 20 bytes for count+magic+zeros?

Reply:
i think so, should have a spec that catches that too